ethereum/ethereum_type.bal

```diff
@@ -0,0 +1,351 @@
+
+    # Creates a filter object, based on filter options, to notify when the state changes (logs).
+    #
+    # + filterOptions - The filter options
+    # + return - If success, return result, else returns EthereumError
+    public function ethNewFilter(json filterOptions) returns Result|EthereumError;
+
+    # Creates a filter in the node, to notify when a new block arrives.
+    #
+    # + return - If success, return result, else returns EthereumError
+    public function ethNewBlockFilter() returns Result|EthereumError;
+
+    # Creates a filter in the node, to notify when new pending transactions arrive.
+    #
+    # + return - If success, return result, else returns EthereumError
+    public function ethNewPendingTransactionFilter() returns Result|EthereumError;
+
+    # Uninstalls a filter with given id.
+    #
+    # + filterId - The filter id
+    # + return - If success, return result, else returns EthereumError
+    public function uninstallEthFilter(string filterId) returns Result|EthereumError;
+
+    # Get an array of logs which occurred since last poll.
+    #
+    # + filterId - The filter id
+    # + return - If success, return result, else returns EthereumError
+    public function getEthFilterChanges(string filterId) returns Result|EthereumError;
+
+    # Get an array of all logs matching filter with given id.
+    #
+    # + filterId - The filter id
+    # + return - If success, return result, else returns EthereumError
+    public function getEthFilterLogs(string filterId) returns Result|EthereumError;
+
+    # Get an array of all logs matching a given filter object.
+    #
+    # + filterOptions - The filter options
+    # + return - If success, return result, else returns EthereumError
+    public function getEthLogs(json filterOptions) returns Result|EthereumError;
+
+    # Get the hash of the current block, the seedHash, and the boundary condition to be met ("target").
+    #
+    # + return - If success, return result, else returns EthereumError
+    public function getEthWork() returns Result|EthereumError;
+
+    # Used for submitting a proof-of-work solution.
+    #
+    # + nonce - The nonce found (64 bits)
+    # + powHash - The header's pow-hash (256 bits)
+    # + mixDigest - The mix digest (256 bits)
+    # + return - If success, return result, else returns EthereumError
+    public function submitEthWork(string nonce, string powHash, string mixDigest) returns Result|EthereumError;
+
+    # Generic method for Ethereum RPC.
+    #
+    # + method - The method name
+    # + params - The parameter array
+    # + return - If success, return result, else returns EthereumError.
+    public function remoteProcedureCall(string method, json params) returns Result|EthereumError;
+};
+
+# Ethereum error.
+#
+# + message - Error message.
+# + cause - The error which caused the Ethereum error.
+# + statusCode - The status code.
+public type EthereumError record {
```

Comment:
No need to have special type called EthereumError. Return `error` from all the functions.